Tools/clinic/clinic.py

```diff
@@ -4906,22 +4906,27 @@ def bad_node(self, node):
         key = f"{parameter_name}_as_{c_name}" if c_name else parameter_name
         self.function.parameters[key] = p
 
-    def parse_converter(self, annotation):
-        if (isinstance(annotation, ast.Constant) and
-            type(annotation.value) is str):
-            return annotation.value, True, {}
-
-        if isinstance(annotation, ast.Name):
-            return annotation.id, False, {}
-
-        if not isinstance(annotation, ast.Call):
-            fail("Annotations must be either a name, a function call, or a string.")
-
-        name = annotation.func.id
-        symbols = globals()
-
-        kwargs = {node.arg: eval_ast_expr(node.value, symbols) for node in annotation.keywords}
-        return name, False, kwargs
+    KwargDict = dict[str | None, Any]
+
+    def parse_converter(
+        self, annotation: ast.expr | None
```

Comment:
The `self` argument isn't actually used in this function, so we could possibly move this function out of the class body and into the global scope. Or we could make it a staticmethod. Or we could leave it as it is.

Reply:
I'd say either static method or status quo.

Reply:
Let's go for static method. It'll probably be a while until we touch this again, so let's exploit this PR to make that improvement.